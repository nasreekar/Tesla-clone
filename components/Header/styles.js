import { StyleSheet } from "react-native"

const styles = StyleSheet.create({
    container: {
        position: "absolute", // to always display above the components
        top: '5%',
        zIndex: 1,
        flexDirection: 'row',
        justifyContent: 'space-between',
        width: '100%',
        paddingHorizontal: 16
    },
    logo: {
        width: 100,
        height: 25,
        resizeMode: 'contain'
    },
    menu: {
        width: 25,
        height: 25
    }
})

export default styles