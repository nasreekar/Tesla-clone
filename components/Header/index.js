import React from 'react'
import styles from './styles'
import { View, Image } from 'react-native'

const Header = (props) => {
    return (
        <View style={styles.container}>
            <Image testID="Tesla logo" style={styles.logo} source={require('../../assets/images/logo.png')}></Image>
            <Image testID="Menu Icon" style={styles.menu} source={require('../../assets/images/menu.png')}></Image>
        </View>
    )

}

export default Header