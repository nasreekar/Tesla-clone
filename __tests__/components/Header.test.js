import React from 'react';
import { render } from "@testing-library/react-native";
import Header from '../../components/Header';

beforeEach(() => {
    jest.useFakeTimers();
  });

describe('<Header />', () => {
    it('renders correctly', () => {
        const { toJSON } = render(<Header />);
        expect(toJSON()).toMatchSnapshot();
      });

    it('contains Tesla logo', () => {
        const { getByTestId } = render(<Header />);
        getByTestId('Tesla logo')
    });

    it('contains Menu Icon', () => {
        const { getByTestId } = render(<Header />);
        getByTestId('Menu Icon')
    });
});