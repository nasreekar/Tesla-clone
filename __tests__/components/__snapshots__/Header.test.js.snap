// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 16,
      "position": "absolute",
      "top": "5%",
      "width": "100%",
      "zIndex": 1,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/logo.png",
      }
    }
    style={
      Object {
        "height": 25,
        "resizeMode": "contain",
        "width": 100,
      }
    }
    testID="Tesla logo"
  />
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/menu.png",
      }
    }
    style={
      Object {
        "height": 25,
        "width": 25,
      }
    }
    testID="Menu Icon"
  />
</View>
`;
